app/scripts/controllers/network/provider-api-tests/shared-tests.js

```diff
@@ -1257,138 +1436,161 @@ export function testsForRpcMethodSupportingBlockParam(
+            comms.mockNextBlockTrackerRequest();
+            comms.mockRpcCall({
+              request: requests[0],
+              response: { result: 'first result' },
+            });
+            comms.mockRpcCall({
+              request: requests[1],
+              response: { result: 'second result' },
+            });
 
-          const results = await withInfuraClient(({ makeRpcCallsInSeries }) =>
-            makeRpcCallsInSeries(requests),
-          );
+            const results = await withClient(
+              { type: providerType },
+              ({ makeRpcCallsInSeries }) => makeRpcCallsInSeries(requests),
+            );
 
-          expect(results).toStrictEqual(['first result', 'second result']);
-        });
+            expect(results).toStrictEqual(['first result', 'second result']);
+          },
+        );
       });
 
       it('makes an additional request to Infura if the given block number matches the latest block number', async () => {
-        await withMockedInfuraCommunications(async (comms) => {
-          const request = {
-            method,
-            params: buildMockParamsWithBlockParamAt(blockParamIndex, '0x100'),
-          };
-
-          // The first time a block-cacheable request is made, the latest
-          // block number is retrieved through the block tracker first. This
-          // also happens within the retry-on-empty middleware (although the
-          // latest block is cached by now).
-          comms.mockNextBlockTrackerRequest({ blockNumber: '0x100' });
-          comms.mockInfuraRpcCall({
-            request,
-            response: { result: 'the result' },
-          });
+        await withMockedCommunications(
+          { type: providerType },
+          async (comms) => {
+            const request = {
+              method,
+              params: buildMockParamsWithBlockParamAt(blockParamIndex, '0x100'),
+            };
 
-          const result = await withInfuraClient(({ makeRpcCall }) =>
-            makeRpcCall(request),
-          );
+            // The first time a block-cacheable request is made, the latest
+            // block number is retrieved through the block tracker first. This
+            // also happens within the retry-on-empty middleware (although the
+            // latest block is cached by now).
+            comms.mockNextBlockTrackerRequest({ blockNumber: '0x100' });
+            comms.mockRpcCall({
+              request,
+              response: { result: 'the result' },
+            });
 
-          expect(result).toStrictEqual('the result');
-        });
+            const result = await withClient(
+              { type: providerType },
+              ({ makeRpcCall }) => makeRpcCall(request),
+            );
+
+            expect(result).toStrictEqual('the result');
+          },
+        );
       });
 
       it('makes an additional request to Infura if the given block number is less than the latest block number', async () => {
```

Comment:
(last one) Another reference to Infura here that should be using `providerType`

Reply:
Fixed in [66aa9c1](https://github.com/MetaMask/metamask-extension/pull/16337/commits/66aa9c1b1c4961ba571948aea065608cb7517087).